cwms-data-api/src/main/java/cwms/cda/api/BinaryTimeSeriesController.java

```diff
@@ -0,0 +1,344 @@
+                            + "otherwise specified). If this field is not specified, "
+                            + "the default time zone of UTC shall be used."),
+                    @OpenApiParam(name = BEGIN, required = true, description = "The start of the time window"),
+                    @OpenApiParam(name = END, description = "The end of the time window. If specified the binaries associated with all times from start to end (inclusive) is retrieved."),
+                    @OpenApiParam(name = VERSION_DATE, description = "The version date for the time series.  If not specified, the maximum version date is used."),
+                    @OpenApiParam(name = Controllers.MIN_ATTRIBUTE, type = Long.class, description = "The minimum attribute value. If not specified, no minimum value is used."),
+                    @OpenApiParam(name = Controllers.MAX_ATTRIBUTE, type = Long.class, description = "The maximum attribute value. If not specified, no maximum value is used."),
+
+            },
+            responses = {
+                    @OpenApiResponse(status = STATUS_200,
+                            content = {
+                                    @OpenApiContent(type = Formats.JSONV2, from = BinaryTimeSeries.class)
+                            }
+                    )},
+            tags = {TAG}
+    )
+    @Override
+    public void getAll(Context ctx) {
+        String office = ctx.queryParam(OFFICE);
+        String tsId = ctx.queryParam(NAME);
+
+        ZonedDateTime beginZdt = queryParamAsZdt(ctx, BEGIN);
+        ZonedDateTime endZdt = queryParamAsZdt(ctx, END);
+        ZonedDateTime versionZdt = queryParamAsZdt(ctx, VERSION_DATE);
+        boolean maxVersion = true;
+        Long minAttr = ctx.queryParamAsClass(Controllers.MIN_ATTRIBUTE, Long.class).getOrDefault(null);
+        Long maxAttr = ctx.queryParamAsClass(Controllers.MAX_ATTRIBUTE, Long.class).getOrDefault(null);
+
+        String binTypeMask = ctx.queryParamAsClass(BINARY_TYPE_MASK, String.class).getOrDefault(DEFAULT_BIN_TYPE_MASK);
+
+        boolean retrieveBinary = true;
+
+        String formatHeader = ctx.header(Header.ACCEPT);
+        ContentType contentType = Formats.parseHeaderAndQueryParm(formatHeader, "");
+        try (Timer.Context ignored = markAndTime(GET_ALL)) {
+            DSLContext dsl = getDslContext(ctx);
+            TimeSeriesBinaryDao dao = getDao(dsl);
+
+            BinaryTimeSeries binaryTimeSeries = dao.retrieve(office, tsId, binTypeMask,
+                    beginZdt == null? null: beginZdt.toInstant(),
+                    endZdt== null? null: endZdt.toInstant(),
+                    versionZdt==null? null: versionZdt.toInstant(),
+                    maxVersion,
+                    retrieveBinary,
+                    minAttr, maxAttr);
+
+            ctx.contentType(contentType.toString());
+
+            String result = Formats.format(contentType, binaryTimeSeries);
```

Comment:
I don't see the corresponding add to JsonV2's format service

Reply:
yeah. Its probably missing and needs to be added.